rules/windows/persistence_sysmon_wmi_event_subscription.toml

```diff
@@ -0,0 +1,89 @@
+note = """## Triage and analysis.
+
+### Investigating Suspicious WMI Event Subscription
+
+WMI (Windows Management Instrumentation) is one of the main Windows components for administration. Because of its
+capability and variety of operations that can be executed, it is an attractive vector for attackers, which can abuse the
+Event Subscription functionality to establish persistence, and in some cases, to escalate privileges to SYSTEM.
+
+To enable the usage of this mechanism, three components are needed: A Event Filter, an Event Consumer, and a binding to connect the first two components.
+
+This rule will monitor the creation of the binding, as it needs to be created to abuse the Event Subscription.
+
+#### Possible investigation steps:
+
+- Identify the account that performed the action.
+- Confirm whether the account owner is aware of the operation.
+- Investigate other alerts related to the user in the last 48 hours.
+- Identify the filter (`winlog.event_data.Filter`) and the consumer (`winlog.event_data.Consumer`).
+- Investigate the created Event filter:
+    - Search for Sysmon event 19 in the host and filter by the filter's name.
+    - Run the `Get-WMIObject -Namespace root/Subscription -Class __EventFilter` command.
+- Investigate the created Event Consumer:
+    - Search for Sysmon event 20 in the host and filter by the consumer's name.
+    - Run the `Get-WMIObject -Namespace root/Subscription -Class __EventConsumer` command.
+
+### False Positive Analysis
+
+- The mechanism can be used legitimately by Microsoft and third party programs, analyze the command line for suspicious
+patterns and potentially malicious activities.
+
+### Response and Remediation
+
+- Initiate the incident response process based on the outcome of the triage.
+- Isolate the involved host to prevent further post-compromise behavior.
+- If the triage identified malware, search the environment for additional compromised hosts.
+- Remove the components related to the malicious event subscription.
+    - Guidance on removing these components can be found in this [blog post](https://medium.com/threatpunter/detecting-removing-wmi-persistence-60ccbb7dff96).
+"""
+references = [
+  "https://www.blackhat.com/docs/us-15/materials/us-15-Graeber-Abusing-Windows-Management-Instrumentation-WMI-To-Build-A-Persistent%20Asynchronous-And-Fileless-Backdoor-wp.pdf",
+  "https://medium.com/threatpunter/detecting-removing-wmi-persistence-60ccbb7dff96",
+]
+risk_score = 47
+rule_id = "e72f87d0-a70e-4f8d-8443-a6407bc34643"
+severity = "medium"
+tags = ["Elastic", "Host", "Windows", "Threat Detection", "Persistence"]
+timestamp_override = "event.ingested"
+type = "eql"
+
+query = '''
+sequence by host.id, user.name with maxspan=1m
+  [any where event.dataset : "windows.sysmon_operational" and event.code : "20" and winlog.event_data.Operation : "Created"]
+  [any where event.dataset : "windows.sysmon_operational" and event.code : "21" and winlog.event_data.Operation : "Created"]
```

Comment:
This will trigger on _any_ filter creation then _any_ binding - which could relate a binding event to an unrelated filter. There is also no guarantee that the same user will create both.

We could make the rule solely on the `21` event and rename the rule to `WMI Persistence via Consumer Binding`. The binding is always the critical component to subscribing to the data for this attack anyway. This would also get around the 1m time constraint, avoiding low-and-slow approaches (and be more performant).


```suggestion
any where event.dataset : "windows.sysmon_operational" and event.code : "21" and winlog.event_data.Operation : "Created"

// to limit to consumer types, add a similar line and replace CommandLineEventConsumer
// and winlog.event_data.Consumer : "*subscription:CommandLineEventConsumer*"
```

Reply:
I've converted this one to a sequence based on feedback from Samir, a sequence can make the triage easier by not having to do aditional queries, wdyt?

Reply:
>This will trigger on any filter creation then any binding - which could relate a binding event to an unrelated filter

Because of that alone, we cannot sequence this. If we had the ability to join on sequences with functions, we could leverage `between`, but in this case, I do not believe it is supported

Reply:
if we want to avoid sequences, I would suggest to use only event 20 instead of 21, 20 contains the command or script to be executed and will help to triage and exclude FPs : 

![image](https://user-images.githubusercontent.com/64742097/159708329-389dc2ce-b196-4a5c-8f16-cf744a0d091c.png)

21 gives the analyst no clues if its malicious or not, just  a name of filter and consumer, they will have to run another query, thinking about the analyst experience during triage time is part of improving user experience.

Reply:
We could potentially have separate rules for them. The consumer is def more verbose, but the binding is what activates it. We could even limit the consumers to just [ActiveScriptEventConsumer](https://docs.microsoft.com/en-us/windows/win32/wmisdk/activescripteventconsumer) and [CommandLineEventConsumer](https://docs.microsoft.com/en-us/windows/win32/wmisdk/commandlineeventconsumer),

Reply:
Wfm, thanks @Samirbous 👍 

Let's change this to the 21 event and other feedback, and we should be good to go